app/src/main/java/com/android/unio/ui/navigation/NavigationAction.kt

```diff
@@ -18,6 +17,9 @@ open class NavigationAction(private val navController: NavHostController) {
    * @param screen The screen to navigate to
    */
   open fun navigateTo(screen: String) {
+    if (navController.currentDestination?.route == screen) {
```

Comment:
This can be rewritten as the shorter `getCurrentRoute() == screen`. Also, it might be useful to add the same check for the TLD `.navigateTo` method.